src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -547,23 +547,23 @@ bool G1CollectedHeap::alloc_archive_regions(MemRegion range) {
   HeapWord* start_address = range.start();
   size_t word_size = range.word_size();
   HeapWord* last_address = range.last();
-  size_t commits = 0;
 
   guarantee(reserved.contains(start_address) && reserved.contains(last_address),
             "MemRegion outside of heap [" PTR_FORMAT ", " PTR_FORMAT "]",
             p2i(start_address), p2i(last_address));
 
   // Perform the actual region allocation, exiting if it fails.
   // Then note how much new space we have allocated.
-  if (!_hrm.allocate_containing_regions(range, &commits, workers())) {
+  regions_committed = 0;
+  uint regions_allocated = 0;
```

Comment:
Why is alloc-count required? Isn't it sth like `ceil(range/region_size)`?

Reply:
This adds implicit assumptions about how the range looks like, i.e. the start address; I think it is nicer if the one allocating the memory returns this value like any other information about the allocation.

Reply:
I think the closure below, `set_region_to_old`, already assumes the start-addr is region-start. In fact, I believe that is an implicit assumption shared by `alloc_archive_regions`, `dealloc_archive_regions`, and `populate_archive_regions_bot_part`.

I'd prefer making it explicit, but YMMV.